staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -912,8 +956,10 @@ func (s *store) getState(ctx context.Context, kv *mvccpb.KeyValue, key string, v
 		state.meta.ResourceVersion = uint64(state.rev)
 
 		if skipTransformDecode {
-			// be explicit that we don't have the object
-			state.obj = nil
+			state.obj = reflect.New(v.Type()).Interface().(runtime.Object)
+			if err := s.versioner.UpdateObject(state.obj, uint64(state.rev)); err != nil {
+				return nil, err
+			}
```

Comment:
I don't understand this change.  Why is our `getState` function  mutating an object?  The `state.rev` returned from here already has the RV for usage elsewhere, right?

Reply:
If I want to run a precondition check with an RV on an unsafe delete, i need an empty object with an RV set. This does not impact the normal deletion flow though.

Reply:
it's used here https://github.com/kubernetes/kubernetes/pull/127513/commits/b1e40fd2e7bd4649686314ad2e4a6259319fe3d2#diff-592aa733e64bdba1062e9957a7d5b0f4124e7f8102ecb4e4278c7d8215e21af6R116-R119

I don't think it's strictly necessary to close this edge case, i can drop this commit if you want.